mm2src/coins_activation/src/prelude.rs

```diff
@@ -0,0 +1,62 @@
+use coins::{coin_conf, CoinBalance, CoinProtocol};
+use common::mm_ctx::MmArc;
+use common::mm_error::prelude::*;
+use serde_derive::Serialize;
+use serde_json::{self as json, Value as Json};
+use std::collections::HashMap;
+
+#[derive(Debug, Serialize)]
+#[serde(tag = "type", content = "data")]
+pub enum DerivationMethod {
+    /// Legacy iguana's privkey derivation, used by default
+    Iguana,
+    /// HD wallet derivation path, String is temporary here
+    #[allow(dead_code)]
+    HDWallet(String),
+}
```

Comment:
Only one question for me, here this enum is helpful and will allow determining the derivation method. And therefore, I think I can use this part in the Solana pull request to support both Solana for those with no seed bip39 and those with one valid seed bip39.

Considering that SLP is a new integration, maybe it makes sense to do this detection now and use the correct derivation method to have the same public address on the HWs and AtomicDEX? If the input seed is not a bip39, we can use Iguana, and for legacy, we can still use the RPC `electrum` V1?

Reply:
> If the input seed is not a bip39, we can use Iguana, and for legacy, we can still use the RPC electrum V1?

It's possible to do this for Solana because it is a completely new protocol. SLP works on top of BCH so changing the private key derivation method for them means changing it for BCH too. And it doesn't seem possible because we likely have users who used BIP39 seed, but MM2 already derived the privkey using the Iguana method.